components/core/src/clp/DictionaryReader.hpp

```diff
@@ -233,26 +233,29 @@ std::string const& DictionaryReader<DictionaryIdType, EntryType>::get_value(Dict
 }
 
 template <typename DictionaryIdType, typename EntryType>
-EntryType const* DictionaryReader<DictionaryIdType, EntryType>::get_entry_matching_value(
+std::vector<EntryType const*>
+DictionaryReader<DictionaryIdType, EntryType>::get_entry_matching_value(
         std::string const& search_string,
         bool ignore_case
 ) const {
+    std::vector<EntryType const*> entries;
     if (false == ignore_case) {
         for (auto const& entry : m_entries) {
             if (entry.get_value() == search_string) {
-                return &entry;
+                entries.push_back(&entry);
+                return entries; /* early exit for case sensitive branch */
```

Comment:
We usually don't use /* */ unless it's docstring.

Let's move the comment on the line above and use // instead
```suggestion
                // early exit for case sensitive branch
                return entries;
```